aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/Orchestrator.java

```diff
@@ -305,9 +321,12 @@ private Completable startApiSync() {
 
     private void stopApiSyncBlocking() {
         try {
-            stopApiSync()
+            boolean subscribed = stopApiSync()
                 .subscribeOn(startStopScheduler)
-                .blockingAwait(OP_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+                .blockingAwait(adjustedTimeoutSeconds, TimeUnit.SECONDS);
+            if (!subscribed) {
+                throw new TimeoutException("Subscription timed out.");
```

Comment:
The "do online stuff" orchestrator mode is currently called `SYNC_VIA_API`. If we change this language here, we should keep both in sync. Maybe it would make more sense to simply call this `ONLINE` mode? `startOnlineBehaviors()` and `stopOnlineBehaviors()`?